server/src/test/java/org/opensearch/action/admin/cluster/snapshots/get/GetSnapshotsActionTests.java

```diff
@@ -0,0 +1,245 @@
+import org.opensearch.action.support.replication.ClusterStateCreationUtils;
+import org.opensearch.cluster.ClusterState;
+import org.opensearch.cluster.metadata.IndexNameExpressionResolver;
+import org.opensearch.cluster.metadata.RepositoryMetadata;
+import org.opensearch.cluster.service.ClusterApplierService;
+import org.opensearch.cluster.service.ClusterService;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.common.util.concurrent.ThreadContext;
+import org.opensearch.index.Index;
+import org.opensearch.repositories.RepositoriesService;
+import org.opensearch.repositories.RepositoriesServiceTests;
+import org.opensearch.repositories.Repository;
+import org.opensearch.tasks.Task;
+import org.opensearch.tasks.TaskId;
+import org.opensearch.test.OpenSearchTestCase;
+import org.opensearch.test.transport.CapturingTransport;
+import org.opensearch.threadpool.TestThreadPool;
+import org.opensearch.threadpool.ThreadPool;
+import org.opensearch.transport.TransportService;
+import org.opensearch.action.ActionRequestValidationException;
+import org.junit.After;
+import org.junit.Before;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+import static org.opensearch.test.ClusterServiceUtils.createClusterService;
+
+public class GetSnapshotsActionTests extends OpenSearchTestCase {
+    private TransportService transportService;
+    private ClusterService clusterService;
+    private ThreadPool threadPool;
+    private RepositoriesService repositoriesService;
+    private final String[] snapshotsName1 = { "test_snapshot" };
+    private final String[] currSnapshotOnly = { GetSnapshotsRequest.CURRENT_SNAPSHOT };
+    private final String[] snapshotsName2 = { "test_snapshot1", "test_snapshot2" };
+    private final String repoName = "test_repo";
+    private final String indexName = "test_index";
+    private static final String repoType = "internal";
+
+    private GetSnapshotsActionTests.TestTransportGetSnapshotsAction getSnapshotsAction;
+
+    public class TestTransportGetSnapshotsAction extends TransportGetSnapshotsAction {
+        TestTransportGetSnapshotsAction() {
+            super(
+                GetSnapshotsActionTests.this.transportService,
+                GetSnapshotsActionTests.this.clusterService,
+                GetSnapshotsActionTests.this.threadPool,
+                repositoriesService,
+                new ActionFilters(Collections.emptySet()),
+                new GetSnapshotsActionTests.Resolver()
+            );
+        }
+
+        @Override
+        protected void clusterManagerOperation(
+            GetSnapshotsRequest request,
+            ClusterState state,
+            ActionListener<GetSnapshotsResponse> listener
+        ) {
+            ClusterState stateWithIndex = ClusterStateCreationUtils.state(indexName, 1, 1);
```

Comment:
you can create ClusterState directly here with snapshots

Reply:
There are no utils as such available with the snapshots, Did not create one as of now.is it required for this test class? As we are able to mock the other objects as well from here.